mycore-viewer/src/main/typescript/modules/mets/widgets/alto/AltoEditorWidget.ts

```diff
@@ -162,41 +167,50 @@ export class AltoEditorWidget {
 
   }
 
-  private getSortFn(by: string, down: boolean): (x, y) => number {
+  private getSortFn(by: string, down: boolean): (x: HTMLElement, y:HTMLElement) => number {
     let headerIndex = ["action", "pageHeading", "actionHeading", "infoHeading"];
     switch (by) {
       case headerIndex[1]:
-        return (x: JQuery, y: JQuery) => {
-          let order1 = this.idChangeMap.get(x.attr("data-id")).pageOrder;
-          let order2 = this.idChangeMap.get(y.attr("data-id")).pageOrder;
+        return (x: HTMLElement, y: HTMLElement) => {
+          let order1 = this.idChangeMap.get(x.getAttribute("data-id")).pageOrder;
+          let order2 = this.idChangeMap.get(y.getAttribute("data-id")).pageOrder;
           return (down ? 1 : -1) * (order1 - order2);
         };
       case headerIndex[2]:
       case headerIndex[3]:
-        return (x: JQuery, y: JQuery) => {
-          let text1 = jQuery(x.children("td").get(headerIndex.indexOf(by))).text();
-          let text2 = jQuery(y.children("td").get(headerIndex.indexOf(by))).text();
+        return (x: HTMLElement, y: HTMLElement) => {
+          let text1 = Array.from(x.children)
+              .filter(el => el.tagName == 'TD')
+              .map(el => el as HTMLTableDataCellElement)[headerIndex.indexOf(by)].innerText;
+          let text2 = Array.from(y.children)
+              .filter(el => el.tagName == 'TD')
+              .map(el => el as HTMLTableDataCellElement)[headerIndex.indexOf(by)].innerText;
 
           return (down ? 1 : -1) * text1.localeCompare(text2);
         }
     }
     return (x, y) => -1;
   }
 
-  private downArrow = jQuery(`
-<span class='fas fa-caret-down sortArrow'>
-</span> 
-`);
+  private createDownArrow() {
+    const downArrow = document.createElement("span");
+    downArrow.classList.add("fas", "fa-caret-down", "sortArrow");
+    return downArrow;
+  }
+
+  private downArrow = this.createDownArrow();
+
+    private createUpArrow() {
+        const upArrow = document.createElement("span");
+        upArrow.classList.add("fas", "fa-caret-up", "sortArrow");
+        return upArrow;
+    }
 
-  private upArrow = jQuery(`    
-<span class='fas fa-caret-up sortArrow'>
-</span> 
-`);
+  private upArrow = this.createUpArrow();
 
 
   private createHTML() {
```

Comment:
is here a return type for the method missing - I don't know the best practice for TS

Reply:
The return type is derived from the content of the function